@here/harp-datasource-protocol/lib/StyleSetEvaluator.ts

```diff
@@ -0,0 +1,409 @@
+                    if (style._index === undefined) {
+                        const technique = this.createTechnique(style, styleStack);
+                        result.push(technique);
+                        if (style.debug) {
+                            logger.log(
+                                "\n======== style w/ technique =========\nenv:",
+                                JSON.stringify(env.unmap(), undefined, 2),
+                                "\nstyle:",
+                                JSON.stringify(style, this.cleanupStyle, 2),
+                                "\ntechnique:",
+                                JSON.stringify(technique, this.cleanupStyle, 2)
+                            );
+                        }
+                    } else {
+                        result.push(this.m_techniques[style._index]);
+                    }
+                }
+                // stop processing if "final" is set
+                return style.final === true;
+            } else if (this.validate) {
+                logger.warn(
+                    "WARNING: No technique defined in style. Either sub-styles or a " +
+                        "technique must be defined:",
+                    JSON.stringify(style)
+                );
+            }
+        }
+        return false;
+    }
+
+    private createTechnique(style: InternalStyle, styleStack: InternalStyle[]) {
+        const technique = {} as any;
+        technique.name = style.technique;
+        const addAttributes = (currStyle: InternalStyle) => {
+            if (currStyle.renderOrder !== undefined) {
+                technique.renderOrder = currStyle.renderOrder;
+            }
+            if (currStyle.transient !== undefined) {
+                technique.transient = currStyle.transient;
+            }
+            if (currStyle.renderOrderBiasProperty !== undefined) {
+                technique.renderOrderBiasProperty = currStyle.renderOrderBiasProperty;
+            }
+            if (currStyle.labelProperty !== undefined) {
+                technique.label = currStyle.labelProperty;
+            }
+            if (currStyle.renderOrderBiasRange !== undefined) {
+                technique.renderOrderBiasRange = currStyle.renderOrderBiasRange;
+            }
+            if (currStyle.renderOrderBiasGroup !== undefined) {
+                technique.renderOrderBiasGroup = currStyle.renderOrderBiasGroup;
+            }
+            if (currStyle.attr !== undefined) {
+                Object.getOwnPropertyNames(currStyle.attr).forEach(property => {
+                    // check for valid attr keys
+                    // TODO: Do proper validation (somewhere else). See HARP-732
+                    if (this.validate) {
+                        if (property === "technique") {
+                            logger.warn(
+                                "WARNING: technique defined in attr " + "(deprecated):",
```

Comment:
Remove the validation?

Reply:
Agreed removed validation completly as it's of no use.